src/structures.rs

```diff
@@ -109,4 +127,38 @@ impl Timetable {
         }
         result
     }
+
+    fn footpaths(stops: &Vec<Stop>, stop_indices: &HashMap<String, u32>) -> Vec<Vec<Footpath>> {
+        let mut result: Vec<Vec<_>> = stops.iter().map(|_| Vec::new()).collect();
+        let mut stop_areas = HashMap::new();
+
+        for stop in stops {
+            if let Some(ref parent) = stop.parent_station {
+                if stop.location_type == 1 || stop.location_type == 0 {
+                    let children = stop_areas.entry(parent).or_insert(Vec::new());
+                    children.push(stop.id.to_owned())
+                }
+            }
+        }
+
+        for (parent, children) in stop_areas {
+            for (child_a, child_b) in children.iter().cartesian_product(&children) {
+                let index_a = *stop_indices
+                    .get(child_a)
+                    .expect(&format!("Missing child station {}", child_a));
+                let index_b = *stop_indices
+                    .get(child_b)
+                    .expect(&format!("Missing child station {}", child_b));
+
+                if index_b != index_a {
```

Comment:
really minor, but the check can be done before the indices get no ? (I think it add some meaning about what you want to test).
or add a `.filter(|a, b| a == b)` ?